apps/meteor/tests/end-to-end/api/02-channels.js

```diff
@@ -1938,6 +1938,62 @@ describe('[Channels]', function () {
 			});
 		});
 
+		it(`should successfully convert a channel to a team when the channel's name is sent as parameter`, (done) => {
+			updatePermission('create-team', ['admin']).then(() => {
+				updatePermission('edit-room', ['admin']).then(() => {
+					request
+						.post(api('teams.convertToChannel'))
+						.set(credentials)
+						.send({ teamName: this.newChannel.name })
+						.expect(200)
+						.expect((res) => {
+							expect(res.body).to.have.a.property('success', true);
+						})
+						.then(() => {
+							request
```

Comment:
this Promise nesting and the previous one with the permissions update is wrong in so many levels 🙈 I see this has been the "standard" on this particular file, but we shouldn't propagate bad code even if it is the current "standard"..

this is conceptually wrong since the `Promise` API was designed to actually mitigate callback hell.. so a correct usage of the `Promise` API should look something like:

```js
updatePermission()
    .then(() => updatePermission())
    .then(() => request
        .post(api('teams.convertToChannel'))
        .set(credentials)
        // ....
    ).then(() => request
        .post(api('teams.convertToTeam'))
        .set(credentials)
        // ....
    )
```

as you can see, everything is done in a **single** level of nesting.

**BUT**, you should also not do this, since `async/await` is available for tests and it is the recommended way to write sequential requests, so please refactor at least the new tests to use `async/await`.